src/cryptonote_core/tx_pool.cpp

```diff
@@ -976,7 +1005,14 @@ namespace cryptonote
   bool tx_memory_pool::have_tx_keyimg_as_spent(const crypto::key_image& key_im) const
   {
     CRITICAL_REGION_LOCAL(m_transactions_lock);
-    return m_spent_key_images.end() != m_spent_key_images.find(key_im);
+    bool spent = false;
+    const auto found = m_spent_key_images.find(key_im);
+    if (found != m_spent_key_images.end())
+    {
+      for (const crypto::hash& tx_hash : found->second)
+        spent |= m_blockchain.txpool_tx_matches_category(tx_hash, relay_category::broadcasted);
```

Comment:
This is an internal (to `tx_pool`) function but I am still mulling over the expected behavior. `do_not_relay` wasn't thought of being private, yet I wouldn't expect setting `do_not_relay` to have _less_ privacy than if I had left that field false. That's basically whats driving my thoughts on this comment and on the one about about ZMQ.

Reply:
Crap, I'm thinking reverting `do_not_relay` is the best option (at least here). Maintaining privacy with that flag means broadcasting a mined block over Tor. Will have to re-think some of these checks, yet again.

Reply:
This now uses `relay_category::legacy`. This category returns false for `local` and `stem` transactions but still returns true for `do_not_relay`. I couldn't think of a better name. Hiding `do_not_relay` txes are funky because peers will notice them in a block which is not notified in a private manner.